src/Gaffer/ScriptNode.cpp

```diff
@@ -533,6 +534,48 @@ void ScriptNode::paste( Node *parent )
 	}
 }
 
+void handleReconnectionOfOutputPlug( const ScriptNode *script, const DependencyNode *dependencyNode, const Plug *outputPlug )
+{
+	// for plugs that hold output plugs we need to recurse
+	if(outputPlug->typeId() == Plug::staticTypeId())
```

Comment:
While we're doing that, let's add a test case to demonstrate reconnection-of-a-child.